ui/components/app/confirmations/alerts/confirm-alert-modal/confirm-alert-modal.tsx

```diff
@@ -0,0 +1,186 @@
+  onSubmit: () => void;
+  isConfirmed: boolean;
+}) {
+  const t = useI18nContext();
+  return (
+    <>
+      <Button
+        block
+        onClick={onCancel}
+        size={ButtonSize.Lg}
+        variant={ButtonVariant.Secondary}
+        data-testid="confirm-alert-modal-cancel-button"
+      >
+        {t('cancel')}
+      </Button>
+      <Button
+        variant={ButtonVariant.Primary}
+        onClick={onSubmit}
+        size={ButtonSize.Lg}
+        data-testid="confirm-alert-modal-submit-button"
+        disabled={!isConfirmed}
+        danger
+      >
+        {t('confirm')}
+      </Button>
+    </>
+  );
+}
+
+function ConfirmDetails({
+  onAlertLinkClick,
+}: {
+  onAlertLinkClick?: () => void;
+}) {
+  const t = useI18nContext();
+  return (
+    <>
+      <Box alignItems={AlignItems.center} textAlign={TextAlign.Center}>
+        <Text variant={TextVariant.bodySm}>
+          {t('confirmAlertModalDetails')}
+        </Text>
+        <ButtonLink
+          paddingTop={5}
+          paddingBottom={5}
+          size={ButtonLinkSize.Inherit}
+          textProps={{
+            variant: TextVariant.bodyMd,
+            alignItems: AlignItems.flexStart,
+          }}
+          as="a"
+          onClick={onAlertLinkClick}
+          target="_blank"
+          rel="noopener noreferrer"
+          data-testid={'confirm-alert-modal-review-all-alerts'}
+        >
+          <Icon
+            name={IconName.SecuritySearch}
+            size={IconSize.Inherit}
+            marginLeft={1}
+          />
+          {t('alertModalReviewAllAlerts')}
+        </ButtonLink>
+      </Box>
+    </>
+  );
+}
+
+export function ConfirmAlertModal({
+  alertKey,
+  onCancel,
+  onClose,
+  onSubmit,
+  ownerId,
+}: ConfirmAlertModalProps) {
+  const t = useI18nContext();
+  const { alerts, isAlertConfirmed } = useAlerts(ownerId);
+
+  const unconfirmedAlerts = alerts.filter(
+    (alert) => alert.field && !isAlertConfirmed(alert.key),
+  );
+  const selectedAlert = alerts.find((alert) => alert.key === alertKey);
+  const hasUnconfirmedAlerts = unconfirmedAlerts.length > 0;
+
+  const [confirmCheckbox, setConfirmCheckbox] = useState<boolean>(false);
+  // if there are multiple alerts, show the multiple alert modal
+  const [multipleAlertModalVisible, setMultipleAlertModalVisible] =
+    useState<boolean>(hasUnconfirmedAlerts);
+
+  const handleCloseMultipleAlertModal = useCallback(() => {
+    setMultipleAlertModalVisible(false);
+  }, []);
+
+  const handleOpenMultipleAlertModal = useCallback(() => {
+    setMultipleAlertModalVisible(true);
+  }, []);
+
+  const handleConfirmCheckbox = useCallback(() => {
+    setConfirmCheckbox(!confirmCheckbox);
+  }, [confirmCheckbox, selectedAlert]);
+
+  if (!selectedAlert) {
+    return null;
+  }
+
+  if (multipleAlertModalVisible) {
+    return (
+      <MultipleAlertModal
+        alertKey={alertKey}
+        ownerId={ownerId}
+        onFinalAcknowledgeClick={handleCloseMultipleAlertModal}
```

Comment:
onFinalAcknowledgeClick is handled as the regular onClose. It does seem redundant that the user will acknowledge each alert individually and then need to acknowledge the final checkbox again after. maybe we could consider checking the final box if all others are checked cc: @SayaGT 

current behavior:
https://github.com/MetaMask/metamask-extension/assets/20778143/da7b3915-492a-4fd7-9c5d-981ef805ba9b

Reply:
It is happening like this because in the storybook it shows the full behaviour of the component, but if the user already acknowledges the alert using the Alert row component it would display only the `ConfirmAlertModal` component.